### DISCUSSION TITLE
Select store based on props (does it makes sense at all? 🤔)

### DISCUSSION THREAD

**Discussion by AlexandrHoroshih** [effector/reflect#20] [Ideas]
Does it make sense to have dynamic store selector? 🤔

Like this:
```ts
const Name = reflect({
  view: Input,
  bind: {
    foo: $data,
  },
  bindSelector: {
    value: (props) => props.isLastName ? $lastName : $fullName // only stores allowed
  }
})

// somewhere inside reflect...
useStore((props) => props.isLastName ? $lastName : $fullName))
```

`effector-react` supports dynamic store selection like that and sometimes people from effector community do use this option, so, maybe, it makes sense to have similiar feature in effector-reflect

**Comment by zerobias** [comment]
Are there any cases that are impossible or very difficult (with a lot of boilerplate code) to solve otherwise? I would offer to show how the implementation of such a component would look like without this functionality, for clarity

The main reason useStore supports this is because of the inevitable dynamism of react itself, which forces it to support this functionality because useStore is called from within a component
